src/Microsoft.OData.Core/Uri/NodeToStringBuilder.cs

```diff
@@ -527,35 +528,35 @@ private ResourceRangeVariableReferenceNode GetResourceRangeVariableReferenceNode
         /// <returns>The translated string with $it binding.</returns>
         private string BindNode(QueryNode node, ResourceRangeVariable filterClauseRangeVariable)
         {
-            BinaryOperatorNode binaryNode = node as BinaryOperatorNode;
-            InNode inNode = node as InNode;
-            AnyNode anyNode = node as AnyNode;
-            SingleValueFunctionCallNode singleValueFunctionCallNode = node as SingleValueFunctionCallNode;
-            SingleValueOpenPropertyAccessNode singleValueOpenPropertyAccessNode = node as SingleValueOpenPropertyAccessNode;
-
-            if (binaryNode != null)
-            {
-                return BindBinaryOperatorNode(binaryNode, filterClauseRangeVariable);
-            }
-            else if (inNode != null)
-            {
-                return BindInNode(inNode, filterClauseRangeVariable);
-            }
-            else if (anyNode != null)
-            {
-                return BindAnyNode(anyNode, filterClauseRangeVariable);
-            }
-            else if (singleValueFunctionCallNode != null)
+            switch (node.Kind)
             {
-                return BindSingleValueFunctionCallNode(singleValueFunctionCallNode, filterClauseRangeVariable);
-            }
-            else if (singleValueOpenPropertyAccessNode != null)
-            {
-                return BindSingleValueOpenPropertyAccess(singleValueOpenPropertyAccessNode, filterClauseRangeVariable);
-            }
-            else
-            {
-                return this.TranslateNode(node);
+                // $expand=Items($filter=$it/Product eq 'Some Product')
+                case QueryNodeKind.BinaryOperator:
+                    BinaryOperatorNode binaryNode = node as BinaryOperatorNode;

```

Comment:
>node as BinaryOperatorNode; [](start = 52, length = 27)

actually, we should use static cast if we already check the node.Kind.

BinaryOperatorNode binaryNode = (BinaryOperatorNode)node;

same for others